chainer/dataset/convert.py

```diff
@@ -139,3 +145,150 @@ def _concat_arrays_with_padding(arrays, padding):
             result[(i,) + slices] = src
 
     return result
+
+
+class ConcatWithAsyncTransfer(object):
+
+    """Interface to concatenate data and transfer them to GPU asynchronously.
+
+    It enables to transfer next batch of input data to GPU while GPU is
+    running kernels for training using current batch of input data.
+
+    Args:
+        stream (cupy.cuda.Stream): CUDA stream. If ``None``, a stream is
+            automatically created on the first call. Data transfer operation
+            is launched acynchrnously using the stream.
+    """
+
+    def __init__(self, stream=None):
+        self.stream = stream
```

Comment:
I think `stream` and `device` should not be modified by (i.e., exposed to) users, so it's better to name them like `_stream` and `_device`.

Reply:
The name of the variables were changed.